packages/kolibri-tools/lib/cli.js

```diff
@@ -0,0 +1,297 @@
+  )
+  .option('-f , --file <file>', 'Set custom file which lists plugins that should be built')
+  .option(
+    '-p, --plugins <plugins...>',
+    'An explicit comma separated list of plugins that should be built',
+    list,
+    []
+  )
+  .option(
+    '--pluginPaths <pluginPaths...>',
+    'An explicit comma separated list of explicit file paths to plugins',
+    list,
+    []
+  )
+  .option('-m, --multi', 'Run using multiple cores to improve build speed', false)
+  .action(function(mode, options) {
+    const { fork } = require('child_process');
+    const buildLogging = logger.getLogger('Kolibri Build');
+    const modes = {
+      DEV: 'dev',
+      PROD: 'prod',
+      I18N: 'i18n',
+      CLEAN: 'clean',
+      STATS: 'stats',
+    };
+    const modeMaps = {
+      c: modes.CLEAN,
+      clean: modes.CLEAN,
+      d: modes.DEV,
+      dev: modes.DEV,
+      development: modes.DEV,
+      p: modes.PROD,
+      prod: modes.PROD,
+      production: modes.PROD,
+      i: modes.I18N,
+      i18n: modes.I18N,
+      internationalization: modes.I18N,
+      s: modes.STATS,
+      stats: modes.STATS,
+    };
+    if (typeof mode !== 'string') {
+      cliLogging.error('Build mode must be specified');
+      process.exit(1);
+    }
+    mode = modeMaps[mode];
+    if (!mode) {
+      cliLogging.error('Build mode invalid value');
+      program.help();
+      process.exit(1);
+    }
+    const multi = options.multi || process.env.KOLIBRI_BUILD_MULTI;
+    const bundleData = readWebpackJson({
+      pluginFile: options.file,
+      plugins: options.plugins,
+      pluginPaths: options.pluginPaths,
+    });
+    if (!bundleData.length) {
+      cliLogging.log('No valid bundle data was returned from the plugins specified');
+      process.exit(1);
+    }
+    const buildModule = {
+      [modes.PROD]: 'production.js',
+      [modes.DEV]: 'webpackdevserver.js',
+      [modes.I18N]: 'i18n.js',
+      [modes.STATS]: 'bundleStats.js',
+    }[mode];
+
+    const modulePath = path.resolve(__dirname, buildModule);
+
+    function spawnWebpackProcesses({ completionCallback = null, persistent = true } = {}) {
+      const numberOfBundles = bundleData.length;
+      let currentlyCompiling = numberOfBundles;
+      // The way we are binding this callback to the webpack compilation hooks
+      // it seems to miss this on first compilation, so we will only use this for
+      // watched builds where rebuilds are possible.
+      function startCallback() {
+        currentlyCompiling += 1;
+      }
+      function doneCallback() {
+        currentlyCompiling -= 1;
+        if (currentlyCompiling === 0) {
+          buildLogging.info('All builds complete!');
+          if (completionCallback) {
+            completionCallback(bundleData);
+          }
+        }
+      }
+      const children = [];
+      for (let index = 0; index < numberOfBundles; index++) {
+        if (multi) {
```

Comment:
This works well on my machine! Building takes noticeably shorter, and makes my computer noticeably hotter :)

If you wanted to prevent all threads from kicking off simultaneously, you could maybe keep a count `numBuildsRunning` and then do something like

```javascript
function build() { /* create child process */ }

function buildEventually() {
  if (numBuildsRunning > numCores - 1) {
    setTimeout(75, buildEventually);
  } else {
    build();
  }
}

buildEventually();
```